sources/advertisements/VRM New Core/Controllers/VRMTimeoutController.swift

```diff
@@ -6,35 +6,50 @@ import PlayerCore
 
 final class VRMTimeoutController {
     
-    let softTimeoutTimerFactory: () -> Cancellable
-    let hardTimeoutTimerFactory: () -> Cancellable
+    typealias OnFire = (()->())
+    typealias TimerFactory = (@escaping OnFire) -> Cancellable
+    
+    let dispatch: (PlayerCore.Action) -> Void
+    let softTimeoutTimerFactory: TimerFactory
+    let hardTimeoutTimerFactory: TimerFactory
     
     private var softTimeoutTimer: Cancellable?
```

Comment:
Make this an optional `struct Timer` with a cancel method?

Reply:
We can't, because for tests we have to use `TimerMock`